libs/vm-modules/tests/unit/ml/ml_model_tests.cpp

```diff
@@ -412,26 +474,26 @@ TEST_F(VMModelTests, model_add_dense_relu)
 
 TEST_F(VMModelTests, model_add_conv1d_noact)
 {
-  TestValidLayerAdding(R"(model.add("conv1d", 10u64, 10u64, 10u64, 10u64);)",
-                       IGNORE_CHARGE_ESTIMATION);
+  TestValidThreeDimLayerAdding(R"(model.add("conv1d", 10u64, 10u64, 10u64, 10u64);)",
+                               IGNORE_CHARGE_ESTIMATION);
 }
 
 TEST_F(VMModelTests, model_add_conv1d_relu)
 {
-  TestValidLayerAdding(R"(model.add("conv1d", 10u64, 10u64, 10u64, 10u64, "relu");)",
-                       IGNORE_CHARGE_ESTIMATION);
+  TestValidThreeDimLayerAdding(R"(model.add("conv1d", 10u64, 10u64, 10u64, 10u64, "relu");)",
+                               IGNORE_CHARGE_ESTIMATION);
 }
 
 TEST_F(VMModelTests, model_add_conv2d_noact)
 {
-  TestValidLayerAdding(R"(model.add("conv2d", 10u64, 10u64, 10u64, 10u64);)",
-                       IGNORE_CHARGE_ESTIMATION);
+  TestValidFourDimLayerAdding(R"(model.add("conv2d", 10u64, 10u64, 10u64, 10u64);)",
+                              IGNORE_CHARGE_ESTIMATION);
 }
 
 TEST_F(VMModelTests, model_add_conv2d_relu)
 {
-  TestValidLayerAdding(R"(model.add("conv2d", 10u64, 10u64, 10u64, 10u64, "relu");)",
-                       IGNORE_CHARGE_ESTIMATION);
+  TestValidFourDimLayerAdding(R"(model.add("conv2d", 10u64, 10u64, 10u64, 10u64, "relu");)",
```

Comment:
I'm not sure how these tests relate to the rest of the PR, but they're useful to have!

Reply:
previously it wasn't necessary to compile model to test adding the layers, as a result a little bit of test fixture editing was needed; other than that it's just a re-branding of an existing test